web/crux-ui/src/components/products/versions/deployments/instances/edit-instance-json.tsx

```diff
@@ -0,0 +1,181 @@
+import JsonEditor from '@app/components/shared/json-editor-dynamic-module'
+import { IMAGE_WS_REQUEST_DELAY } from '@app/const'
+import { CANCEL_THROTTLE, useThrottling } from '@app/hooks/use-throttleing'
+import { CompleteContainerConfig, ContainerConfig, InstanceContainerConfig, UniqueKeyValue } from '@app/models-config'
+import { fold } from '@app/utils'
+import { completeContainerConfigSchema } from '@app/validation'
+import clsx from 'clsx'
+import { useCallback, useEffect, useReducer } from 'react'
+import { v4 as uuid } from 'uuid'
+
+interface EditInstanceJsonProps {
+  disabled?: boolean
+  className?: string
+  config: InstanceContainerConfig
+  disabledContainerNameEditing?: boolean
+  onPatch: (config: Partial<InstanceContainerConfig>) => void
+  onParseError?: (err: Error) => void
+}
+
+type EditImageJsonActionType = 'config-change' | 'set-content'
+
+type EditImageJsonAction = {
+  type: EditImageJsonActionType
+  content?: CompleteContainerConfig
+  config?: ContainerConfig
+}
+
+const DEFAULT_CONFIG = completeContainerConfigSchema.getDefault() as any as CompleteContainerConfig
+
+const keyValueArrayToJson = (envs: UniqueKeyValue[]): Record<string, string> =>
+  fold(envs, {}, (prev, it) => {
+    if (!prev[it.key]) {
+      prev[it.key] = it.value
+    }
+
+    return prev
+  })
+
+const imageConfigToCompleteContainerConfig = (
+  currentConfig: CompleteContainerConfig,
+  imageConfig: ContainerConfig,
+): CompleteContainerConfig => {
+  if (!imageConfig) {
+    return currentConfig ?? DEFAULT_CONFIG
+  }
+
+  const config: CompleteContainerConfig = {
+    ...(imageConfig.config ?? currentConfig ?? DEFAULT_CONFIG),
+    name: imageConfig.name ?? currentConfig.name,
+    environment: currentConfig?.environment ?? {},
+    capabilities: currentConfig?.capabilities ?? {},
+    secrets: currentConfig?.secrets ?? {},
+  }
+
+  if (imageConfig.environment) {
+    config.environment = keyValueArrayToJson(imageConfig.environment)
+  }
+
+  if (imageConfig.capabilities) {
+    config.capabilities = keyValueArrayToJson(imageConfig.capabilities)
+  }
+
+  return config
+}
+
+const reducer = (state: CompleteContainerConfig, action: EditImageJsonAction): CompleteContainerConfig => {
+  const { type } = action
+
+  if (type === 'config-change') {
+    return imageConfigToCompleteContainerConfig(state, action.config)
+  }
+  if (type === 'set-content') {
```

Comment:
switch or ifelse would be better, nope? 🤔

Reply:
Originall it was an if-else but the linter did not like it because each branch returns.

Reply:
Hmm, same as know, but okey.